manifests/init.pp

```diff
@@ -0,0 +1,43 @@
+# @summary Manage /etc/hosts entries
+#
+# @param hosts A hash of hosts entries to manage
+#
+# @param default_hosts A hash of hosts entries to manage by default
+#
+# @param purge unmanaged host resources
+#
+# @param manage_fqdn manage the fqdn entry
+#
+class hosts (
+  Optional[Hash[String[1], Hash[String[1], Any]]] $hosts = undef,
+  Optional[Hash[String[1], Hash[String[1], Any]]] $default_hosts = undef,
```

Comment:
Can you switch to empty hash as defaults?
```suggestion
  Hash[String[1], Hash[String[1], Any]] $hosts = {},
  Hash[String[1], Hash[String[1], Any]] $default_hosts = {},
```

Reply:
I dislike using `{}` as a default because it means setting the param to `undef` will cause things to blow up, which I dislike.

Reply:
well, it looks like I already used `{}` in the acceptance test, so I guess it won't hurt to simply the code...

Reply:
yes :) And why would you set it to undef anyways?
We also covered that in in [style guide](https://voxpupuli.org/docs/reviewing_pr/):
> Are there new parameters with datatype Hash or Array? If possible, they should default to empty Hash or Array instead of undef. You can also enforce the datastructure like Array[String[1]] (that’s recommended if possible)